lib/Gocdb_Services/Site.php

```diff
@@ -1408,6 +1408,11 @@ public function addAPIAuthEntity(\Site $site, \User $user, $newValues) {
         if ($type == 'X509' && !preg_match("/^(\/[A-Za-z]+=[a-zA-Z0-9\/\-\_\s\.,'@:\/]+)*$/", $identifier)) {
             throw new \Exception("Invalid x509 DN");
         }
+        
+        //If the entity is of type OIDC subject, do a more thorough check again
+        if ($type == 'OIDC Subject' && !preg_match("/^([a-z0-9]{8}\-[a-z0-9]{4}\-[a-z0-9]{4}\-[a-z0-9]{4}\-[a-z0-9]{12})$/", $identifier)) {
```

Comment:
I was asked about whether this format would change - for completeness, this is the standard format for a Universally Unique identifier (UUID): [https://en.wikipedia.org/wiki/Universally_unique_identifier](https://en.wikipedia.org/wiki/Universally_unique_identifier)

Reply:
Looking at the standard format it should only contain characters a-f not a-z, so I think the following is better to avoid a 'false positive'

```suggestion
        if ($type == 'OIDC Subject' && !preg_match("/^([a-f0-9]{8}\-[a-f0-9]{4}\-[a-f0-9]{4}\-[a-f0-9]{4}\-[a-f0-9]{12})$/", $identifier)) {
```

Reply:
Good point - missed that! All characters in the UUID are Hex, so this makes sense.